### DISCUSSION TITLE
How to  generated a SRS  through MPC for plonk?

### DISCUSSION THREAD

**Discussion by zhengjunhe** [Consensys/gnark#944] [Q&A]
Now I could only generate a SRS  for plonk via test.NewKZGSRS(ccs). There are packages named mpcsetup for groth16,but no packages to do mpc setup for plonk.

How could I generated a SRS  through MPC for plonk?

@yelhousni and @ivokub 
Thanks in advance.

**Comment by ivokub** [comment]
What curve are you interested in? For BN254 we have https://github.com/Consensys/gnark-ignition-verifier

Namely, for PLONK we do not need per-circuit trusted setup, but it is suitable to use already existing trusted setup for KZG. Different organizations have held ceremonies for different curves. See also discussion https://github.com/Consensys/gnark/discussions/922

Unfortunately, we do not have MPC implementation for generating trusted setup for KZG. I think Groth16 mpcsetup package could be repurposed for that effect, but it is not planned right now. Will create an issue though if someone would like to tackle in the future :)

**Reply by ivokub** [reply]
Created issue https://github.com/Consensys/gnark/issues/945 for tracking.

**Comment by zhengjunhe** [comment]
BN254 is ok for me now.
Thanks and I will try the srs you provided for plonk.

Ivo Kubjas ***@***.***>于2023年12月4日 周一21:44写道：
